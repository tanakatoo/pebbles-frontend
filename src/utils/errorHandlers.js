import { rest } from 'msw'

let loginResponse = {
    token: 'abcdefg',
    user: {
        id: 2,
        username: 'hello',
        role: 'regular',
        premium_join_date: '',
        premium_end_date: '',
        free_trial_start_date: '',
    }
}



export const errorHandlers = [

    //login
    rest.post('http://localhost:3001/auth/login', (req, res, ctx) => {
        const failLogin = {
            "error": {
                "message": "INVALID_CREDENTIALS",
                "status": 401
            }
        }
        console.log(res(ctx.json([failLogin])))
        return res(ctx.json([failLogin]));
    }),

    rest.post('http://localhost:3001/email', (req, res, ctx) => {
        console.log('MOCK error /email');
        const noData = {
            "error": {
                "message": "no data",
                "status": 400
            }
        }
        return res(ctx.json(noData));
    }),

]