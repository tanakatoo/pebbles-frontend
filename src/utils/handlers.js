import { rest } from 'msw'

let loginResponse = {
    token: 'abcdefg',
    user: {
        id: 2,
        username: 'hello',
        role: 'regular',
        premium_join_date: '',
        premium_end_date: '',
        free_trial_start_date: '',
    }
}

const contactsResponse =
    [
        {
            "user_id": 1,
            "username": "ktoo",
            "avatar": "2.jpg"
        },
        {
            "user_id": 15,
            "username": "june",
            "avatar": "4.jpg"
        }
    ]

const blockedResponse = [
    {
        "blocked_user_id": 3,
        "username": "blockMe",
        "avatar": "9.jpg"
    }
]
const mainSearchResults = [
    {
        "username": "hello2",
        "avatar": "9.jpg",
        "name": null,
        "about": null,
        "premium_acct_id": null,
        "end_date": null
    },
    {
        "username": "helloo",
        "avatar": "14.jpg",
        "name": null,
        "about": null,
        "premium_acct_id": null,
        "end_date": null
    }
]

const changePassword = "completed"

const newMessage = [
    {
        "toavatar": "9.jpg",
        "to": "hello",
        "fromavatar": "",
        "from": "hello"
    }
]

const continuingMsg = [
    {
        "toavatar": "9.jpg",
        "fromavatar": "2.jpg",
        "from": "ktoo",
        "to": "hello",
        "from_user_id": 1,
        "to_user_id": 2,
        "msg": "the most recent one reply to second one",
        "sent_at": "2023-06-09 15:30:00",
        "read": true
    },
    {
        "toavatar": "2.jpg",
        "fromavatar": "9.jpg",
        "from": "hello",
        "to": "ktoo",
        "from_user_id": 2,
        "to_user_id": 1,
        "msg": "2nd most recent one reply",
        "sent_at": "2023-06-08 15:30:00",
        "read": true
    },
    {
        "toavatar": "9.jpg",
        "fromavatar": "2.jpg",
        "from": "ktoo",
        "to": "hello",
        "from_user_id": 1,
        "to_user_id": 2,
        "msg": "first message",
        "sent_at": "2023-06-07 15:30:00",
        "read": true
    }
]

const latestMessages = [
    {
        "toavatar": "9.jpg",
        "fromavatar": "4.jpg",
        "from": "june",
        "to": "hello",
        "from_user_id": 15,
        "to_user_id": 2,
        "msg": "testing\n",
        "sent_at": "2023-07-15 20:57:44.804634",
        "read": false
    },
    {
        "toavatar": "9.jpg",
        "fromavatar": "2.jpg",
        "from": "ktoo",
        "to": "hello",
        "from_user_id": 1,
        "to_user_id": 2,
        "msg": "the most recent one reply to second one",
        "sent_at": "2023-06-09 15:30:00",
        "read": true
    }
]

const returnUser = [
    {
        "id": 2,
        "username": "hello",
        "name": null,
        "avatar": "9.jpg",
        "country_en": null,
        "country_ja": null,
        "city_en": null,
        "city_ja": null,
        "state_en": null,
        "state_ja": null,
        "gender": "female",
        "about": null,
        "free_trial_start_date": null,
        "study_buddy_bio": "just chatting",
        "study_buddy_purpose": "i don't know why",
        "native_language": "Japanese",
        "learning_language": "English",
        "language_level": "Beginner",
        "time_zone": "NewYork",
        "age_range": "26-35",
        "study_buddy_active": true,
        "study_buddy_activate_date": "2023-07-13T04:00:00.000Z"
    }
]

export const handlers = [

    //login
    rest.post('http://localhost:3001/auth/login', (req, res, ctx) => {

        console.log(res(ctx.json(loginResponse)))
        return res(ctx.json(loginResponse));
    }),

    rest.post('http://localhost:3001/auth/logout', (req, res, ctx) => {

        console.log('MOCK /logout')
        window.localStorage.setItem('token', null)
        return res(ctx.json());
    }),


    // Called via <ExploreCommunity type="studyBuddy" />
    rest.get('http://localhost:3001/study-buddies/1', (req, res, ctx) => {
        console.log('MOCK /study-buddies/1');

        return res(ctx.json([]));
    }),


    rest.post('http://localhost:3001/auth/register', (req, res, ctx) => {
        console.log('MOCK /auth/register');

        return res(ctx.json(loginResponse));
    }),


    rest.post('http://localhost:3001/email', (req, res, ctx) => {
        console.log('MOCK /email');

        return res(ctx.json('sents'));
    }),


    rest.get('http://localhost:3001/users/contacts', (req, res, ctx) => {
        console.log('MOCK /users/contacts');

        return res(ctx.json(contactsResponse));
    }),

    rest.get('http://localhost:3001/users/blocked', (req, res, ctx) => {
        console.log('MOCK /users/blocked');

        return res(ctx.json(blockedResponse));
    }),

    rest.get('http://localhost:3001/users/search', (req, res, ctx) => {
        console.log('MOCK /users/search');

        return res(ctx.json(mainSearchResults));
    }),

    rest.post('http://localhost:3001/auth/change-password', (req, res, ctx) => {
        console.log('MOCK /auth/change-password');

        return res(ctx.json(changePassword));
    }),

    //for continuing conversation
    rest.get('http://localhost:3001/messages/:username', (req, res, ctx) => {
        console.log('MOCK /messages/:username');
        console.log(req.params)
        return res(ctx.json(continuingMsg));
    }),

    //get latest message from all users
    rest.get('http://localhost:3001/messages', (req, res, ctx) => {
        console.log('MOCK /messages');
        return res(ctx.json(latestMessages));
    }),

    //get latest message from all users
    rest.get('http://localhost:3001/users/saved-users', (req, res, ctx) => {
        console.log('MOCK /users/saved-users');
        return res(ctx.json(returnUser));
    }),
]