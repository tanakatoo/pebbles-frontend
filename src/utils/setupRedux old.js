
import { configureStore } from '@reduxjs/toolkit';
import rootReducer from "../reducers/rootReducer"
import { persistStore, persistReducer, FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER } from 'redux-persist';

import storage from 'redux-persist/lib/storage'
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'

export const persistConfig = {
    key: 'root', // Key prefix for the persisted state
    storage, // Storage mechanism (e.g., local storage)
    stateReconciler: autoMergeLevel2, // Specify which parts of the state to persist
    whitelist: ['profile', 'langFont']
};

export const persistedReducer = persistReducer(persistConfig, rootReducer);
export const store = configureStore({
    reducer: persistedReducer,
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
            serializableCheck: {
                ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER]
            }
        })
})
export const persistedStore = persistStore(store)