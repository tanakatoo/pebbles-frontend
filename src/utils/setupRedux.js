import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
// As a basic setup, import your same slice reducers
import { persistStore, persistReducer, FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER } from 'redux-persist';
import { PersistGate } from 'redux-persist/integration/react'
import { ListSlice } from "../reducers/rootReducer";
import rootReducer from "../reducers/rootReducer"
import storage from 'redux-persist/lib/storage'
import storeSetup from "./storeSetup"
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'

// const persistConfig = {
//     key: 'root', // Key prefix for the persisted state
//     storage, // Storage mechanism (e.g., local storage)
//     stateReconciler: autoMergeLevel2, // Specify which parts of the state to persist
//     whitelist: ['profile', 'langFont']
// };
// const persistedReducer = persistReducer(persistConfig, rootReducer);

// const persistedStore = persistStore(store)
// const root = ReactDOM.createRoot(document.getElementById('root'));


export function renderWithProviders(
    ui,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = storeSetup(preloadedState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {

        return <Provider store={store}>{children}</Provider>;
    }

    // Return an object with the store and all of RTL's query functions
    return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}