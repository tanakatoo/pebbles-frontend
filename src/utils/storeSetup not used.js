import { combineReducers, configureStore } from '@reduxjs/toolkit'

import rootReducer from '../reducers/rootReducer'
// import userReducer from '../features/users/userSlice'

// Create the root reducer separately so we can extract the RootState type
// const rootReducer = combineReducers({
//     user: userReducer
// })

export const setupStore = preloadedState => {
    return configureStore({
        reducer: rootReducer,
        preloadedState
    })
}