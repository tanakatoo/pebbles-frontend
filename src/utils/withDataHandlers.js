import { rest } from 'msw'

let studybuddyResponse = [
    {
        "id": 2,
        "username": "hello",
        "name": null,
        "avatar": "9.jpg",
        "country_en": null,
        "country_ja": null,
        "city_en": null,
        "city_ja": null,
        "state_en": null,
        "state_ja": null,
        "gender": "female",
        "about": null,
        "free_trial_start_date": null,
        "study_buddy_bio": "just chatting",
        "study_buddy_purpose": "i don't know why",
        "native_language": "Japanese",
        "learning_language": "English",
        "language_level": "Beginner",
        "time_zone": "NewYork",
        "age_range": "26-35",
        "study_buddy_active": true,
        "study_buddy_activate_date": "2023-07-13T04:00:00.000Z",
        "study_buddy_types": [
            "StudyBuddy",
            "Volunteer"
        ]
    }
]

console.log('in data handlers', studybuddyResponse)

export const withDataHandlers = [

    //get some study buddies
    rest.get('http://localhost:3001/study-buddies/1', (req, res, ctx) => {
        console.log('MOCK with data /study-buddies/1')
        console.log(studybuddyResponse)
        return res(ctx.json(studybuddyResponse));
    }),

]