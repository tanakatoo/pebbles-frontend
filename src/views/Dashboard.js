import React from 'react'
import CTA from '../components/common/CTA'
import usePageText from "../hooks/usePageText"
import { useSelector } from 'react-redux'
import StatisticsCard from '../components/dashboard/StatisticsCard'
import { AwesomeCheck } from '../styles/Icons'
import { Button } from '../components/button/Button'
import ExploreCommunity from '../components/dashboard/ExploreCommunity'


function Dashboard() {
    const [pageText, lang] = usePageText('dashboard')
    const profile = useSelector(state => state.profile.profile)
    console.log('in dashboa')
    return (
        // <Protected>
        <div>
            {lang === "JA"
                ?
                profile.premium_join_date && !profile.premium_end_date ?
                    <>
                        <CTA msg={pageText.CTA_MSG_PREMIUM} msgBtn={pageText.CTA_MSG_BTN_PREMIUM} btnLink={pageText.CTA_MSG_BTN_LINK_PREMIUM} />
                    </>
                    :
                    profile.free_trial_start_date ?
                        <>
                            <CTA msg={pageText.CTA_MSG_NO_PREMIUM} msgBtn={pageText.CTA_MSG_BTN_NO_PREMIUM} btnLink={pageText.CTA_MSG_BTN_LINK_NO_PREMIUM} />
                        </>

                        :
                        <>
                            <CTA msg={pageText.CTA_MSG_NO_TRIAL} msgBtn={pageText.CTA_MSG_BTN_NO_TRIAL} btnLink={pageText.CTA_MSG_BTN_LINK_NO_TRIAL} />



                            <div className='mx-4'>
                                <div className='flex flex-col items-center justify-center mb-8 md:mb-12 gap-2'>
                                    <h2 className="text-primary-dark font-medium text-mobile-page-header text-center">{pageText.STAT}</h2>
                                    <div className='flex flex-row flex-wrap items-center my-4 gap-4 justify-around grow container w-full'>
                                        <StatisticsCard numUsers={pageText.STAT_NUM_LEVEL_UP} text={pageText.STAT_LEVEL_UP} viaPercent='via-30%' />
                                        <StatisticsCard numUsers={pageText.STAT_NUM_UPGRADED} text={pageText.STAT_UPGRADED} viaPercent={'via-60%'} />
                                        <StatisticsCard numUsers={pageText.STAT_NUM_STARTED} text={pageText.STAT_STARTED} viaPercent={'via-20%'} />
                                    </div>
                                </div>
                            </div>
                            {profile.premium_join_date && !profile.premium_end_date ?
                                null :
                                profile.free_trial_start_date ?
                                    <div className='flex flex-col '>
                                        <h2 className="bg-secondary py-2 text-white font-medium text-mobile-page-header text-center">{pageText.PREMIUM_BENEFITS}</h2>
                                        <div className='bg-secondary-background p-4 flex flex-col items-center'>
                                            <p className='mb-2 md:max-w-[500px]'>{pageText.PREMIUM_BLURB}</p>

                                            <ol>
                                                <li className='py-2 md:py-1'><span className='text-success me-2'><AwesomeCheck /></span> <span className=''>{pageText.CHECK1}</span></li>
                                                <li className='py-2 md:py-1'><span className='text-success me-2'><AwesomeCheck /></span> <span className=''>{pageText.CHECK2}</span></li>
                                                <li className='py-2 md:py-1'><span className='text-success me-2'><AwesomeCheck /></span> <span className=''>{pageText.CHECK3}</span></li>
                                                <li className='py-2 md:py-1'><span className='text-success me-2'><AwesomeCheck /></span> <span className=''>{pageText.CHECK4}</span></li>
                                            </ol>
                                        </div>
                                        <Button btnText={pageText.PREMIUM_BTN} />
                                    </div >
                                    :
                                    <div className='flex flex-col bg-secondary-background pb-8 md:pb-12'>
                                        <h2 className="bg-secondary py-2 text-white font-medium text-mobile-page-header px-4 md:text-center">{pageText.TRIAL_BENEFITS}</h2>
                                        <div className='p-4  flex flex-col items-center justify-center'>
                                            <p className='mb-4 md:max-w-[800px]'>{pageText.TRIAL_BLURB}</p>
                                            <ol className=''>
                                                <li className='py-2 md:py-1'><span className='text-success me-2'><AwesomeCheck /></span> <span className=''>{pageText.TRIAL_CHECK1}</span></li>
                                                <li className='py-2 md:py-1'><span className='text-success me-2'><AwesomeCheck /></span> <span className=''>{pageText.TRIAL_CHECK2}</span></li>
                                                <li className='py-2 md:py-1'><span className='text-success me-2'><AwesomeCheck /></span> <span className=''>{pageText.TRIAL_CHECK3}</span></li>
                                                <li className='py-2 md:py-1'><span className='text-success me-2'><AwesomeCheck /></span> <span className=''>{pageText.TRIAL_CHECK4}</span></li>
                                                <li className='py-2 md:py-1'><span className='text-success me-2'><AwesomeCheck /></span> <span className=''>{pageText.TRIAL_CHECK5}</span></li>
                                            </ol>
                                        </div>
                                        <div className='flex justify-center mt-6'>
                                            <Button btnText={pageText.TRIAL_BTN} />
                                        </div>
                                    </div >

                            }
                        </>
                :
                <>
                    <CTA msg={pageText.CTA_MSG} msgBtn={pageText.CTA_MSG_BTN} btnLink={pageText.CTA_MSG_BTN_LINK} />
                </>

            }

            <h2 className="py-2 my-8 text-primary-dark font-medium text-mobile-page-header px-4 text-center ">{pageText.EXPLORE_COMMUNITY}</h2>
            <div className='flex flex-col pt-2 pb-12 bg-primary-super-light'>
                <div className='flex items-center justify-center flex-1'>
                    <ExploreCommunity type="studyBuddy" />
                </div>
            </div >

            <div className='flex flex-col pt-2 pb-12 bg-regional-background'>
                <div className='flex items-center justify-center flex-1'>
                    <ExploreCommunity type="infoCenter" />
                </div>
            </div >
            <div className='flex flex-col pt-2 pb-12 bg-accent-very-light'>
                <div className='flex items-center justify-center flex-1'>
                    <ExploreCommunity type="languageTown" />
                </div>
            </div >
            <div className='flex flex-col pt-2 pb-12 bg-marketplace-background'>
                <div className='flex items-center justify-center flex-1'>
                    <ExploreCommunity type="marketplace" />
                </div>
            </div >
        </div >
        // </Protected>
    )
}

export default Dashboard