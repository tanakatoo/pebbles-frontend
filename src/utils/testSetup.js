// import { persistedStore, store } from './setupRedux';
import { render } from '@testing-library/react'
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { setupTestStore } from './setupRedux';


const renderWithProviders = (ui, options = {}) => {
    const [store, persistedStore] = setupTestStore(options.preloadedState);

    const AllProviders = ({ children }) => {
        return (
            <Provider store={store}>
                <PersistGate loading={null} persistor={persistedStore}>
                    {children}
                </PersistGate>
            </Provider>
        )
    };

    return { store, ...render(ui, { wrapper: AllProviders, ...options }) };
}


// "re-export everything"
export * from '@testing-library/react';

export { renderWithProviders };