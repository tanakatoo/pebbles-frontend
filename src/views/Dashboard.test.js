import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { renderWithProviders } from '../utils/testSetup';
import Dashboard from './Dashboard';
import { waitFor } from '../utils/testSetup';
import { withDataHandlers } from '../utils/withDataHandlers';
import { server } from '../utils/server'
import { act } from 'react-dom/test-utils';
import { fireEvent } from '../utils/testSetup';
import { screen } from '../utils/testSetup';


// Mock window.scrollTo()
window.scrollTo = jest.fn();
afterAll(() => {
    jest.clearAllMocks();
});


describe('Dashboard', () => {
    // test('renders without crashing', () => {
    //     renderWithProviders(
    //         <MemoryRouter>
    //             <Dashboard />
    //         </MemoryRouter>
    //     );
    // });

    // test('matches snapshot', async () => {
    //     const { asFragment } = renderWithProviders(
    //         <MemoryRouter>
    //             <Dashboard />
    //         </MemoryRouter>
    //     );

    //     expect(asFragment()).toMatchSnapshot();
    // });

    // test('displays no study buddies', async () => {
    //     const { getByTestId, getByLabelText, getByText } = renderWithProviders(
    //         <MemoryRouter>
    //             <Dashboard />
    //         </MemoryRouter>
    //     );

    //     // Wait for pageText to be populated
    //     await waitFor(() => {
    //         expect(getByTestId('dashStudyBuddyNoData')).not.toBeEmptyDOMElement();
    //     });

    //     expect(getByText(/Learn how to use Pebbles/)).toBeInTheDocument();
    //     //There should not be any study buddies
    //     expect(getByText(/No study buddies yet! /)).toBeInTheDocument();
    // });

    test('displays 1 study buddy when click goes to study buddy page', async () => {
        server.use(...withDataHandlers)

        const { getByTestId, getByLabelText, getByText } =
            renderWithProviders(
                <MemoryRouter>
                    <Dashboard />
                </MemoryRouter>
            );

        // Wait for pageText to be populated
        await waitFor(() => {
            expect(getByTestId('dashStudyBuddyWithData')).not.toBeEmptyDOMElement();
        });

        expect(getByText(/Volunteer/)).toBeInTheDocument();
        //There should not be any study buddies
        expect(getByText(/hello/)).toBeInTheDocument();

        // Click on See All for study buddies

        await act(() => {
            fireEvent.click(getByText(/See all/));
        })

        // Wait for pageText to be populated
        await waitFor(() => {
            expect(getByTestId('pageTitlepageText')).not.toBeEmptyDOMElement();
        });

    });


});