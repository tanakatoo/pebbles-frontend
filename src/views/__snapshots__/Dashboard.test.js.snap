// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard displays study buddies 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex w-full bg-primary text-white px-4 py-3 mb-8 justify-center items-center"
    >
      <p
        class="mr-8 font-medium"
      >
        Learn Japanese for free
      </p>
      <a
        href="/study-buddies"
      >
        <button
          class="drop-shadow 
                inline-flex
                items-center
                justify-center
                gap-2
                bg-white 
                text-primary-dark 
                py-2 
                px-4 
                rounded-xl
                disabled:opacity-50 text-mobile-card-header font-medium}
                hover:opacity-75"
          data-testid=""
          type="button"
        >
          <span
            class="font-medium"
          >
            Study buddy board
          </span>
        </button>
      </a>
    </div>
    <h2
      class="py-2 my-2 text-primary-dark font-medium text-mobile-page-header px-4 text-center "
      data-testid="wait-for-pagetext"
    >
      Explore the community
    </h2>
    <p
      class="mb-10 text-center group"
    >
      <a
        class="hover:text-gray-text hover:underline hover:underline-offset-4"
        href="/faq"
      >
        Learn how to use Pebbles
      </a>
       
      <span
        class="group-hover:text-gray-text"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-right-long fa-1x "
          data-icon="arrow-right-long"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-128-128c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l370.7 0-73.4 73.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l128-128z"
            fill="currentColor"
          />
        </svg>
      </span>
    </p>
    <div
      class="flex flex-col pt-2 pb-12 bg-primary-super-light"
    >
      <div
        class="flex items-center justify-center flex-1"
      >
        <div
          class="w-full  flex flex-col"
        >
          <h3
            class="grow px-4 py-2 text-mobile-section-header font-medium}"
          >
            Newest Study Buddies
          </h3>
          <div
            class=" my-24"
          >
            <div
              class="w-full flex justify-center"
            >
              <div
                class="inline-block h-8 w-8 rounded-full animate-spin border-r-4 border-t-4 border-l-4  align-[-0.125em] motion-reduce:animate-[spin_1.5s_linear_infinite]"
                role="status"
              >
                <span
                  class="!absolute !-m-px !h-px !w-px !overflow-hidden !whitespace-nowrap !border-0 !p-0 ![clip:rect(0,0,0,0)]"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col pt-2 pb-12 bg-regional-background"
    >
      <div
        class="flex items-center justify-center flex-1"
      >
        <div
          class="w-full  flex flex-col"
        >
          <h3
            class="grow px-4 py-2 text-mobile-section-header font-medium}"
          >
            Newest from Regional Information Center
          </h3>
          <div
            class="my-12"
          >
            <div
              class="text-center "
            >
              <span>
                Under construction. Please be patient!
              </span>
              <p>
                <a
                  class="hover:text-gray-text hover:underline hover:underline-offset-4"
                  href="/"
                />
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col pt-2 pb-12 bg-accent-very-light"
    >
      <div
        class="flex items-center justify-center flex-1"
      >
        <div
          class="w-full  flex flex-col"
        >
          <h3
            class="grow px-4 py-2 text-mobile-section-header font-medium}"
          >
            Newest from Japanese Town
          </h3>
          <div
            class="my-12"
          >
            <div
              class="text-center "
            >
              <span>
                Under construction. Please be patient!
              </span>
              <p>
                <a
                  class="hover:text-gray-text hover:underline hover:underline-offset-4"
                  href="/"
                />
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col pt-2 pb-12 bg-marketplace-background"
    >
      <div
        class="flex items-center justify-center flex-1"
      >
        <div
          class="w-full  flex flex-col"
        >
          <h3
            class="grow px-4 py-2 text-mobile-section-header font-medium}"
          >
            Newest from Marketplace
          </h3>
          <div
            class="my-12"
          >
            <div
              class="text-center "
            >
              <span>
                Under construction. Please be patient!
              </span>
              <p>
                <a
                  class="hover:text-gray-text hover:underline hover:underline-offset-4"
                  href="/"
                />
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dashboard matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex w-full bg-primary text-white px-4 py-3 mb-8 justify-center items-center"
    >
      <p
        class="mr-8 font-medium"
      >
        Learn Japanese for free
      </p>
      <a
        href="/study-buddies"
      >
        <button
          class="drop-shadow 
                inline-flex
                items-center
                justify-center
                gap-2
                bg-white 
                text-primary-dark 
                py-2 
                px-4 
                rounded-xl
                disabled:opacity-50 text-mobile-card-header font-medium}
                hover:opacity-75"
          data-testid=""
          type="button"
        >
          <span
            class="font-medium"
          >
            Study buddy board
          </span>
        </button>
      </a>
    </div>
    <h2
      class="py-2 my-2 text-primary-dark font-medium text-mobile-page-header px-4 text-center "
      data-testid="wait-for-pagetext"
    >
      Explore the community
    </h2>
    <p
      class="mb-10 text-center group"
    >
      <a
        class="hover:text-gray-text hover:underline hover:underline-offset-4"
        href="/faq"
      >
        Learn how to use Pebbles
      </a>
       
      <span
        class="group-hover:text-gray-text"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-arrow-right-long fa-1x "
          data-icon="arrow-right-long"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-128-128c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l370.7 0-73.4 73.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l128-128z"
            fill="currentColor"
          />
        </svg>
      </span>
    </p>
    <div
      class="flex flex-col pt-2 pb-12 bg-primary-super-light"
    >
      <div
        class="flex items-center justify-center flex-1"
      >
        <div
          class="w-full  flex flex-col"
        >
          <h3
            class="grow px-4 py-2 text-mobile-section-header font-medium}"
          >
            Newest Study Buddies
          </h3>
          <div
            class=" my-24"
          >
            <div
              class="w-full flex justify-center"
            >
              <div
                class="inline-block h-8 w-8 rounded-full animate-spin border-r-4 border-t-4 border-l-4  align-[-0.125em] motion-reduce:animate-[spin_1.5s_linear_infinite]"
                role="status"
              >
                <span
                  class="!absolute !-m-px !h-px !w-px !overflow-hidden !whitespace-nowrap !border-0 !p-0 ![clip:rect(0,0,0,0)]"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col pt-2 pb-12 bg-regional-background"
    >
      <div
        class="flex items-center justify-center flex-1"
      >
        <div
          class="w-full  flex flex-col"
        >
          <h3
            class="grow px-4 py-2 text-mobile-section-header font-medium}"
          >
            Newest from Regional Information Center
          </h3>
          <div
            class="my-12"
          >
            <div
              class="text-center "
            >
              <span>
                Under construction. Please be patient!
              </span>
              <p>
                <a
                  class="hover:text-gray-text hover:underline hover:underline-offset-4"
                  href="/"
                />
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col pt-2 pb-12 bg-accent-very-light"
    >
      <div
        class="flex items-center justify-center flex-1"
      >
        <div
          class="w-full  flex flex-col"
        >
          <h3
            class="grow px-4 py-2 text-mobile-section-header font-medium}"
          >
            Newest from Japanese Town
          </h3>
          <div
            class="my-12"
          >
            <div
              class="text-center "
            >
              <span>
                Under construction. Please be patient!
              </span>
              <p>
                <a
                  class="hover:text-gray-text hover:underline hover:underline-offset-4"
                  href="/"
                />
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col pt-2 pb-12 bg-marketplace-background"
    >
      <div
        class="flex items-center justify-center flex-1"
      >
        <div
          class="w-full  flex flex-col"
        >
          <h3
            class="grow px-4 py-2 text-mobile-section-header font-medium}"
          >
            Newest from Marketplace
          </h3>
          <div
            class="my-12"
          >
            <div
              class="text-center "
            >
              <span>
                Under construction. Please be patient!
              </span>
              <p>
                <a
                  class="hover:text-gray-text hover:underline hover:underline-offset-4"
                  href="/"
                />
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
